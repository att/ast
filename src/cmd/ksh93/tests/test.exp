# Test for test builtin
set pid [spawn $ksh]
expect_prompt

# ==========
log_test_entry
send "test -t 1 && echo PASS || echo FAILED\r"
expect "\r\nPASS\r\n" {
    puts "stdin is a terminal device"
}
expect_prompt

# ==========
# `test -t` without argument should check for stdout
log_test_entry
send "test -t && echo PASS || echo FAILED\r"
expect "\r\nPASS\r\n" {
    puts "stdout is a terminal device"
}
expect_prompt

# Exit shell with ctrl-d
log_test_entry
send [ctrl D]
catch {expect default exp_continue} output
log_debug "EOF output: $output"

catch {wait}
exit 0
