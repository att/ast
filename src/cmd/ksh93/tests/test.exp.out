stdin is a terminal device
stdout is a terminal device
